operator/pkg/service/envoy_config.go

```diff
@@ -43,14 +56,271 @@ func newEnvoyConfigManager(maxRetries int) (*envoyConfigManager, error) {
 }
 
 // getByKey is a wrapper of Store.GetByKey but with concrete Endpoint object
-func (em *envoyConfigManager) getByKey(key string) (*v2.CiliumEnvoyConfig, bool, error) {
+func (em *envoyConfigManager) getByKey(key string) (*ciliumv2.CiliumEnvoyConfig, bool, error) {
 	objFromCache, exists, err := em.store.GetByKey(key)
 	if objFromCache == nil || !exists || err != nil {
 		return nil, exists, err
 	}
-	envoyConfig, ok := objFromCache.(*v2.CiliumEnvoyConfig)
+	envoyConfig, ok := objFromCache.(*ciliumv2.CiliumEnvoyConfig)
 	if !ok {
 		return nil, exists, fmt.Errorf("got invalid object from cache")
 	}
 	return envoyConfig, exists, err
 }
+
+func getEnvoyConfigForService(svc *slim_corev1.Service) (*ciliumv2.CiliumEnvoyConfig, error) {
+	resources, err := getResources(svc)
+	if err != nil {
+		return nil, err
+	}
+	return &ciliumv2.CiliumEnvoyConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       ciliumv2.CECKindDefinition,
+			APIVersion: ciliumv2.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%s", ciliumEnvoyLBPrefix, svc.GetName()),
+			Namespace: svc.GetNamespace(),
+			OwnerReferences: []metav1.OwnerReference{
```

Comment:
Should this set `Controller: true`  and/or `BlockOwnerDeletion`?  It's been a while since I've dealt with operators/controllers, but I think that those normally should be set?

Reply:
I don't think controller: true is should be used here, as normally it's for another intermediate controller/resource such as ReplicaSet. 

I think we can skip BlockOwnerDeletion to have faster cleanup for parent resource (e.g. Service)